### ISSUE TITLE
CVE vulnerabilities in Grpc.Core

### ISSUE THREAD

**Issue by briandunnington** [Azure/azure-functions-durable-extension#2521]
### Description

Microsoft.Azure.WebJobs.Extensions.DurableTask (even the recent v2.10.0) use Grpc.Core v2.46.5 which has three Component Governance vulnerabilites:

CVE-2023-32731
CVE-2023-1428
CVE-2023-32732

Grpc.Core reference comes in via Microsoft.DurableTask.Sidecar.Protobuf which has not been updated for ~8 months. The advice is to update to Grpc.Core 2.52.0 directly, but there is no such version (there is no published version higher than the affected version).

**Comment by cgillum** [comment]
Indeed, it looks like there's no replacement for Grpc.Core v2.46.5, probably because that project has been [deprecated](https://github.com/grpc/grpc/issues/32719).

We'll need to look into replacing that dependency with the newer one owned by Microsoft (Grpc.Net.Client), assuming that it's fully compatible in all the places we need to use it. Currently, Grpc.Net.Client on .NET Framework doesn't support HTTP/2 on Windows 10, as documented [here](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/netstandard.md#net-framework), which may be a big problem for us.

@jviau @fabiocav is this an issue the impacts the Functions host at all? I'm wondering how we go about addressing it.

**Comment by li-thy-um** [comment]
Same issue. Received a lot of CG bugs.

**Comment by isghu210293** [comment]
I have the same issue, is there any update on updating that Grpc.Core.dll dependency
